Source/Core/DolphinQt/SkylanderPortal/SkylanderPortalWindow.cpp

```diff
@@ -0,0 +1,867 @@
+
+  CreateMainWindow();
+
+  connect(&Settings::Instance(), &Settings::EmulationStateChanged, this,
+          &SkylanderPortalWindow::OnEmulationStateChanged);
+
+  installEventFilter(this);
+
+  OnEmulationStateChanged(Core::GetState());
+};
+
+SkylanderPortalWindow::~SkylanderPortalWindow() = default;
+
+void SkylanderPortalWindow::CreateMainWindow()
+{
+  auto* main_layout = new QVBoxLayout();
+
+  auto* checkbox_group = new QGroupBox();
+  auto* checkbox_layout = new QHBoxLayout();
+  checkbox_layout->setAlignment(Qt::AlignHCenter);
+  m_checkbox = new QCheckBox(tr("Emulate Skylander Portal"), this);
+  m_checkbox->setChecked(Config::Get(Config::MAIN_EMULATE_SKYLANDER_PORTAL));
+  connect(m_checkbox, &QCheckBox::toggled, [=](bool checked) { EmulatePortal(checked); });
+  checkbox_layout->addWidget(m_checkbox);
+  checkbox_group->setLayout(checkbox_layout);
+  main_layout->addWidget(checkbox_group);
+
+  auto add_line = [](QVBoxLayout* vbox) {
+    auto* line = new QFrame();
+    line->setFrameShape(QFrame::HLine);
+    line->setFrameShadow(QFrame::Sunken);
+    vbox->addWidget(line);
+  };
+
+  m_group_skylanders = new QGroupBox(tr("Active Portal Skylanders:"));
+  auto* vbox_group = new QVBoxLayout();
+  auto* scroll_area = new QScrollArea();
+
+  for (auto i = 0; i < MAX_SKYLANDERS; i++)
+  {
+    if (i != 0)
+    {
+      add_line(vbox_group);
+    }
+
+    auto* hbox_skylander = new QHBoxLayout();
+    auto* label_skyname = new QLabel(QString(tr("Skylander %1")).arg(i + 1));
+    m_edit_skylanders[i] = new QLineEdit();
+    m_edit_skylanders[i]->setEnabled(false);
+
+    auto* clear_btn = new QPushButton(tr("Clear"));
+    auto* create_btn = new QPushButton(tr("Create"));
+    auto* load_btn = new QPushButton(tr("Load"));
+
+    connect(clear_btn, &QAbstractButton::clicked, this, [this, i]() { ClearSkylander(i); });
+    connect(create_btn, &QAbstractButton::clicked, this, [this, i]() { CreateSkylander(i); });
+    connect(load_btn, &QAbstractButton::clicked, this, [this, i]() { LoadSkylander(i); });
+
+    hbox_skylander->addWidget(label_skyname);
+    hbox_skylander->addWidget(m_edit_skylanders[i]);
+    hbox_skylander->addWidget(clear_btn);
+    hbox_skylander->addWidget(create_btn);
+    hbox_skylander->addWidget(load_btn);
+
+    vbox_group->addLayout(hbox_skylander);
+  }
+
+  m_group_skylanders->setLayout(vbox_group);
+  scroll_area->setWidget(m_group_skylanders);
+  scroll_area->setWidgetResizable(true);
+  m_group_skylanders->setVisible(Config::Get(Config::MAIN_EMULATE_SKYLANDER_PORTAL));
+  main_layout->addWidget(scroll_area);
+  setLayout(main_layout);
+
+  UpdateEdits();
+}
+
+void SkylanderPortalWindow::OnEmulationStateChanged(Core::State state)
+{
+  const bool running = state != Core::State::Uninitialized;
+
+  m_checkbox->setEnabled(!running);
+}
+
+CreateSkylanderDialog::CreateSkylanderDialog(QWidget* parent) : QDialog(parent)
+{
+  setWindowTitle(tr("Skylander Creator"));
+  setObjectName(tr("skylanders_creator"));
```

Comment:
I don't think it makes sense for "skylanders_creator" to be a **tr**anslatable string.